qiskit/algorithms/time_evolvers/time_evolution_result.py

```diff
@@ -27,20 +28,27 @@ class TimeEvolutionResult(AlgorithmResult):
         aux_ops_evaluated (ListOrDict[tuple[complex, complex]] | None): Optional list of
             observables for which expected values on an evolved state are calculated. These values
             are in fact tuples formatted as (mean, standard deviation).
+        observables (ListOrDict[tuple[np.ndarray, np.ndarray]] | None): Optional list of
+            observables for which expected on an evolved state are calculated at each timestep.
+            These values are in fact lists of tuples formatted as (mean, standard deviation).
     """
 
     def __init__(
         self,
         evolved_state: QuantumCircuit,
         aux_ops_evaluated: ListOrDict[tuple[complex, complex]] | None = None,
+        observables: ListOrDict[tuple[np.ndarray, np.ndarray]] | None = None,
```

Comment:
Perhaps we should generalize the other argument instead of introducing this new one?

Reply:
The other way of doing it was (or still is) being deprecated. One of them should go at some point but it will be a different PR.

Reply:
Yeah I would agree that `observable` makes more sense in time evolution than `aux_ops`, which comes from VQE. With the primitives, the evaluation of auxiliary operators at the final state might be obsolete anyways, since you can just submit another estimator within the same session 🙂